modules/javafx.graphics/src/main/java/javafx/scene/image/PixelBuffer.java

```diff
@@ -0,0 +1,220 @@
+     *                                  not have sufficient memory, or if the type of {@code buffer}
+     *                                  and {@code pixelFormat} do not match
+     * @throws NullPointerException     if {@code buffer} or {@code pixelFormat} is {@code null}
+     */
+    public PixelBuffer(int width, int height, T buffer, PixelFormat<T> pixelFormat) {
+        Objects.requireNonNull(buffer, "buffer must not be null.");
+        Objects.requireNonNull(pixelFormat, "pixelFormat must not be null.");
+        if (width <= 0 || height <= 0) {
+            throw new IllegalArgumentException("PixelBuffer dimensions must be positive (w,h > 0)");
+        }
+        switch (pixelFormat.getType()) {
+            case BYTE_BGRA_PRE:
+                if (buffer.capacity() / width / 4 < height) {
+                    throw new IllegalArgumentException("Insufficient memory allocated for ByteBuffer.");
+                }
+                if (!(buffer instanceof ByteBuffer)) {
+                    throw new IllegalArgumentException("PixelFormat<ByteBuffer> requires a ByteBuffer.");
+                }
+                break;
+            case INT_ARGB_PRE:
+                if (buffer.capacity() / width < height) {
+                    throw new IllegalArgumentException("Insufficient memory allocated for IntBuffer.");
+                }
+                if (!(buffer instanceof IntBuffer)) {
+                    throw new IllegalArgumentException("PixelFormat<IntBuffer> requires an IntBuffer.");
+                }
+                break;
+            default:
+                throw new IllegalArgumentException("Unsupported PixelFormat: " + pixelFormat.getType());
+        }
+        this.buffer = buffer;
+        this.width = width;
+        this.height = height;
+        this.pixelFormat = pixelFormat;
+        this.imageRefs = new LinkedList<>();
+    }
+
+    /**
+     * Returns the {@link java.nio.Buffer} of this {@code PixelBuffer}.
+     *
+     * @return the {@link java.nio.Buffer} of this {@code PixelBuffer}
+     */
+    public T getBuffer() {
+        return buffer;
+    }
+
+    /**
+     * Returns the width of this {@code PixelBuffer}.
+     *
+     * @return the width of this {@code PixelBuffer}
+     */
+    public int getWidth() {
+        return width;
+    }
+
+    /**
+     * Returns the height of this {@code PixelBuffer}.
+     *
+     * @return the height of this {@code PixelBuffer}
+     */
+    public int getHeight() {
+        return height;
+    }
+
+    /**
+     * Returns the {@code PixelFormat} of this {@code PixelBuffer}.
+     *
+     * @return the {@code PixelFormat} of this {@code PixelBuffer}
+     */
+    public PixelFormat<T> getPixelFormat() {
+        return pixelFormat;
+    }
+
+    /**
+     * Invokes the specified {@link Callback} method and updates the dirty region of
+     * all {@code WritableImage}s that were created using this {@code PixelBuffer}.
```

Comment:
I think that the summary (first sentence) shouldn't bother saying that the method invokes the callback. What the method does is "updates the dirty region...". What the callback has to do with it is explained in the next sentence.

Reply:
Not sure I agree. Invoking the callback method is a primary purpose of this method.

Reply:
Alright, I don't have a strong preference.

Reply:
Keeping the current version of doc.